arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -356,21 +343,26 @@ public fun <A> resource(block: suspend ResourceScope.() -> A): Resource<A> = blo
  * <!--- KNIT example-resource-06.kt -->
  */
 @ScopeDSL
-public suspend fun <A> resourceScope(action: suspend ResourceScope.() -> A): A {
-  val scope = ResourceScopeImpl()
-  val a: A = try {
-    action(scope)
+public suspend inline fun <A> resourceScope(action: ResourceScope.() -> A): A = with(ResourceScopeImpl()) {
+  try {
+    action()
   } catch (e: Throwable) {
-    val ex = if (e is CancellationException) ExitCase.Cancelled(e) else ExitCase.Failure(e)
-    val ee = withContext(NonCancellable) {
-      scope.cancelAll(ex, e) ?: e
-    }
-    throw ee
-  }
-  withContext(NonCancellable) {
-    scope.cancelAll(ExitCase.Completed)?.let { throw it }
-  }
-  return a
+    cancelAll(ExitCase(e))
+    error("Unreachable, cancelAll should throw the exception passed to it. Please report this bug to the Arrow team.")
+  }.also { cancelAll(Completed) }
```

Comment:
Can we remove `also` here? I'd prefer regular function invocation after `}` if possible.

I try to avoid scoping function everywhere where they might be redundant. I also see you changed a lot of function bodies, to expressions, and rely on `also` instead. I personally feel this really degrades readability.

This is a common occurrence in this PR, I rather not have these unrelated code changes purely for style reasons. It introduces a lot of code churn for no strong reason. Also takes away the focus of this PR.